### PR TITLE
702 subscription list query

### PR DESCRIPTION
**Pull request by ricardovdheijden**
#702  SubscriptionList query

### CODE REVIEW

File: packages/orchestrator-ui-components/src/rtk/endpoints/subscriptionList.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            GraphqlQueryVariables<SubscriptionListItem>
+        >({
+            query: (variables) => ({
+                document: subscriptionListQuery,
+                variables,
+            }),
+            transformResponse: (
+                response: SubscriptionsResult,
+            ): SubscriptionListResponse => {
+                const subscriptions = response.subscriptions.page || [];
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/rtk/endpoints/subscriptionList.ts:72]: More of a general comment but I would map all the logic that converts the graphql result into a SubscriptionList Item (like applying mapGraphQlSubscriptionsResultToSubscriptionListItems) to this function. So you get items you can use directly from the hook

File: packages/orchestrator-ui-components/src/rtk/endpoints/subscriptionListSummary.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            query: (variables) => ({
+                document: subscriptionListSummaryQuery,
+                variables,
+            }),
+            transformResponse: (
+                response: SubscriptionsResult<SubscriptionSummary>,
+            ): SubscriptionListSummaryResponse => {
+                const subscriptions = response.subscriptions.page || [];
+                const pageInfo = response.subscriptions.pageInfo || {};
+
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/rtk/endpoints/subscriptionListSummary.ts:58]: Same comment as above

**Review by DutchBen — APPROVED** [review]

File: packages/orchestrator-ui-components/src/rtk/endpoints/subscriptionListSummary.ts (hunk 2 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/rtk/endpoints/subscriptionListSummary.ts:58]: It is hard to make a good abstraction here. The prop, in this case, `subscriptions` is different in most situations (processes, productBlocks etc)

File: packages/orchestrator-ui-components/src/rtk/endpoints/subscriptionList.ts (hunk 1 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/rtk/endpoints/subscriptionList.ts:72]: I would not move such logic in here. Data from the backend should stay roughly in its shape. SubscriptionListItems are tailored for use in the table, typical view data. Adding that mapper here fully transforms the result from the graphql and makes possible reuse of this endpoint hard.

**Review by tjeerddie — APPROVED** [review]

**Pull request merged**